### PR TITLE
Merge ClusterVersion to Version

### PR DESCRIPTION
**Pull request by vbekiaris**
Expands `Version` to `major.minor` instead of a single byte version value.
Unify `ClusterVersion` to `Version` to simplify development for backwards compatibility.

### CODE REVIEW

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java [hunk 1]
```diff
@@ -234,7 +234,7 @@ public void shutdown() {
     @Override
     public void validateJoinRequest(JoinMessage joinMessage) {
         // check joining member's major.minor version is same as current cluster version's major.minor numbers
-        if (!joinMessage.getVersion().asClusterVersion().equals(node.getClusterService().getClusterVersion())) {
+        if (!joinMessage.getVersion().asVersion().equals(node.getClusterService().getClusterVersion())) {
             throw new VersionMismatchException("Joining node's version " + joinMessage.getVersion() + " is not compatible with "
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java]: Hmm, do we need this asVersion()?

Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java]: Aha, shouldn't it be getMemberVersion().asVersion() to make it less confusing?

File: hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java [hunk 2]
```diff
@@ -268,9 +268,9 @@ public void onClusterVersionChange(ClusterVersion newVersion) {
     }
 
     @Override
-    public boolean isNodeVersionCompatibleWith(ClusterVersion clusterVersion) {
+    public boolean isNodeVersionCompatibleWith(Version clusterVersion) {
         Preconditions.checkNotNull(clusterVersion);
-        return node.getVersion().asClusterVersion().equals(clusterVersion);
+        return node.getVersion().asVersion().equals(clusterVersion);
     }
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java]: Same here? There's probably more places like this.

File: hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java (hunk 1 above)
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java]: 👍 just pushed another commit that renames `JoinMessage.version` to `memberVersion`

**Pull request merged**